nixos/modules/services/networking/thelounge.nix

```diff
@@ -0,0 +1,85 @@
+{ pkgs, lib, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.thelounge;
+  configJsData = "module.exports = " + builtins.toJSON (
+    { private = cfg.private; port = cfg.port; } // cfg.extraConfig
+  );
+in {
+  options.services.thelounge = {
+    enable = mkEnableOption "The Lounge web IRC client";
+
+    home = mkOption {
+      type = types.path;
+      default = "/var/lib/thelounge";
+      description = ''
+        Sets the storage location of The Lounge's user config and log files.
+      '';
+    };
+
+    private = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Make your The Lounge instance private. You will need to configure user
+        accounts by adding entries in <filename>${cfg.home}/users</filename>.
+      '';
+    };
+
+    port = mkOption {
+      type = types.int;
+      default = 9000;
+      description = "TCP port to listen on for http connections.";
+    };
+
+    extraConfig = mkOption {
+      default = {};
+      type = types.attrs;
+      example = {
+        reverseProxy = true;
+        defaults = {
+          name = "Your Network";
+          host = "localhost";
+          port = 6697;
+        };
+      };
+      description = ''
+        The Lounge's <filename>config.js</filename> contents as attribute set (will be
+        converted to JSON to generate the configuration file).
+
+        The options defined here will be merged to the default configuration file.
+        Note: In case of duplicate configuration, options from extraConfig have priority.
+
+        Documentation: https://thelounge.chat/docs/server/configuration
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.users = singleton {
+      name = "thelounge";
+      uid = config.ids.uids.thelounge;
+      description = "The Lounge daemon user";
+      home = cfg.home;
+      createHome = true;
+    };
+
+    systemd.services.thelounge = {
+      description = "The Lounge web IRC client";
+      wantedBy = [ "multi-user.target" ];
+      wants = [ "network-online.target" ];
+      after = [ "network-online.target" ];
```

Comment:
I don't think this service needs `network-online.target`, does it?
Would it not just try to reconnect after some time?

Reply:
It is delaying the service start for no good reason.

Reply:
You can also checkout the last section here: https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/